tests/test_utils.py

```diff
@@ -100,44 +97,128 @@
             image_position=(10.0, 60.0, 0.0),
             image_orientation=(0.0, -1.0, 0.0, -1.0, 0.0, 0.0),
             pixel_spacing=(0.5, 2.0),
-            spacing_between_slices=0.0,
         ),
-        (6.0, 58.0, 0.0),
+        (2.0, 59.0, 0.0),
     ),
     pytest.param(
         dict(
             coordinate=(5.0, 2.0),
             image_position=(10.0, 60.0, 0.0),
             image_orientation=(1.0, 0.0, 0.0, 0.0, -1.0, 0.0),
             pixel_spacing=(0.5, 2.0),
-            spacing_between_slices=0.0,
         ),
-        (11.0, 50.0, 0.0),
+        (12.5, 56.0, 0.0),
     ),
     pytest.param(
         dict(
             coordinate=(5.0, 4.0),
             image_position=(10.0, 60.0, 0.0),
             image_orientation=(1.0, 0.0, 0.0, 0.0, -1.0, 0.0),
             pixel_spacing=(0.5, 2.0),
-            spacing_between_slices=0.0,
         ),
-        (12.0, 50.0, 0.0),
+        (12.5, 52.0, 0.0),
     ),
     pytest.param(
         dict(
             coordinate=(5.0, 4.0),
             image_position=(10.0, 60.0, 30.0),
             image_orientation=(1.0, 0.0, 0.0, 0.0, -1.0, 0.0),
             pixel_spacing=(0.5, 2.0),
-            spacing_between_slices=0.0,
         ),
-        (12.0, 50.0, 30.0),
+        (12.5, 52.0, 30.0),
+    ),
+    # Patient
+    pytest.param(
+        dict(
+            coordinate=(5.0, 4.0),
+            image_position=(10.0, 60.0, 0.0),
+            image_orientation=(1.0, 0.0, 0.0, 0.0, 1.0, 0.0),
+            pixel_spacing=(0.5, 2.0),
+        ),
+        (12.5, 68.0, 0.0),
+    ),
```

Comment:
Your current tests failed to find the error with the z coordinate calculation mentioned above because you always use 0.0 as the third element of `image_position`. I suggest adding some tests with non-zero values here. Also add some where the z component of the two cosine vectors is non-zero. E.g. a sagittal or coronal orientation. Here's an example for a sagittal image (confirmed with SimpleITK and my corrected version of the highdicom function above):

```python
pytest.param(
    dict(
        coordinate=(15.0, 87.0),
        image_position=(-45.0, 35.0, -90.0),
        image_orientation=(0.0, 1.0, 0.0, 0.0, 0.0, -1.0),  # axis-aligned sagittal orientation
        pixel_spacing=(0.5, 0.25),
    ),
    (-45.0, 42.5, -111.75),
)
```